src/test/java/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStepWorkflowTest.java

```diff
@@ -153,6 +154,49 @@ public void evaluate() throws Throwable {
 
     }
 
+    /**
+     * This test verifies that sshAgent step handles that the build agent
+     * disconnects and reconnects during the step execution.
+     */
+    @Issue("JENKINS-59259")
+    @Test
+    public void agentConnectionDropTest() throws Exception {
+
+        story.addStep(new Statement() {
```

Comment:
BTW if you use `then` you can pass a lambda as a body, which is a bit prettier.

Reply:
Great tip, thanks!